rest_framework/schemas/openapi.py

```diff
@@ -209,6 +211,33 @@ def _get_pagination_parameters(self, path, method):
 
         return paginator.get_schema_operation_parameters(view)
 
+    def _map_choicefield(self, field):
+        choices = list(field.choices)
+        if all(isinstance(choice, bool) for choice in choices):
+            type = 'boolean'
+        elif all(isinstance(choice, int) for choice in choices):
+            type = 'integer'
+        elif all(isinstance(choice, (int, float, Decimal)) for choice in choices):  # `number` includes `integer`
+            # Ref: https://tools.ietf.org/html/draft-wright-json-schema-validation-00#section-5.21
+            type = 'number'
+        elif all(isinstance(choice, str) for choice in choices):
+            type = 'string'
+        else:
+            type = None
+
+        mapping = {
+            # The value of `enum` keyword MUST be an array and SHOULD be unique.
+            # Ref: https://tools.ietf.org/html/draft-wright-json-schema-validation-00#section-5.20
+            'enum': list(OrderedDict.fromkeys(choices))  # preserve order and remove duplicates
```

Comment:
Also, would it make sense for us to move the `OrderedDict` call up to the local variable `choices` that we use above?

Reply:
Agreed.

Reply:
Done